src/recipe/recipe.service.ts

```diff
@@ -20,6 +20,21 @@ export class RecipeService {
     private readonly s3Service: S3Service,
   ) {}
 
+  getImageUrlsToUrls(recipe: Recipe): Promise<string[]> {
+    return Promise.all(
+      recipe.imageKeys.map((key) => this.s3Service.getPresignedUrl(key)),
+    );
+  }
+
+  getImageUrlsOfManyRecipes(recipes: Recipe[]): Promise<Recipe[]> {
+    return Promise.all(
+      recipes.map(async (recipe) => {
+        recipe.imageKeys = await this.getImageUrlsToUrls(recipe);
+        return recipe;
+      }),
+    );
```

Comment:
check if you can use spread operator here (I'm not 100% sure):

```suggestion
    return Promise.all(
      recipes.map(async (recipe) => ({ ...recipe, imageKeys: await this.getImageUrlsToUrls(recipe) })),
    );
```